Sprout/View Controllers/LoginViewController.m

```diff
@@ -35,13 +39,85 @@ - (void)viewDidLoad {
     
     self.loginAlert = [UIAlertController alertControllerWithTitle:@"Error during user login." message:@"Please try again!" preferredStyle:(UIAlertControllerStyleAlert)];
     [self.loginAlert addAction:okAction];
-
 }
 
 - (IBAction)dismissKeyboard:(id)sender {
     [self.view endEditing:true];
 }
 
+- (IBAction)continueWithFacebook:(id)sender {
+    //TODO: allow users to create account through Facebook or log in through facebook. rn this is registering user
+    FBSDKLoginManager *loginManager = [FBSDKLoginManager new];
+    [loginManager logOut];
+    [loginManager logInWithPermissions:@[@"public_profile", @"email"]
+                    fromViewController:self
+                               handler:^(FBSDKLoginManagerLoginResult *_Nullable result, NSError *_Nullable error){
+        if (error == nil && !result.isCancelled) {
+            self.graphRequest = [[FBSDKGraphRequest alloc]
+                                 initWithGraphPath:@"/me"
+                                 parameters:@{@"fields":@"first_name,last_name,email,picture"}
+                                 HTTPMethod:@"GET"];
+            //TODO: modify graph request/the following method to get a high-res profile picture
+            [self createUserThroughFB: self.graphRequest];
+        }
+        else{
+            NSLog(@"%@", error.localizedDescription);
+        }
+    }];
+}
+
+- (void)createUserThroughFB:(FBSDKGraphRequest *)request {
+    [request startWithCompletion:^(
+                                   id<FBSDKGraphRequestConnecting> _Nullable connection,
+                                   id _Nullable result,
+                                   NSError *_Nullable error) {
+        if (result) {
+            PFQuery *query = [PFUser query];
+            [query whereKey:@"email" equalTo:result[@"email"]];
+            [query countObjectsInBackgroundWithBlock:^(int count, NSError * _Nullable error) {
+                if (count == 0) {
+                    PFUser *newUser = [PFUser user];
```

Comment:
This gets a little unwieldy when it's all in one method. I'd probably break this into a helper method so it reads more declaratively. 

```
[query countObjectsInBackgroundWithBlock:^(int count, NSError * _Nullable error) {
    if (count == 0) {
        [self createNewUserFromResult:result];
```